lib/ansible/plugins/cache/__init__.py

```diff
@@ -298,18 +299,19 @@ def update(self, key, value):
         self._plugin.set(key, host_cache)
 
 
-class InventoryFileCacheModule(BaseFileCacheModule):
-    """
-    A caching module backed by file based storage.
-    """
-    def __init__(self, plugin_name, timeout, cache_dir):
-
-        self.plugin_name = plugin_name
-        self._timeout = timeout
+class InventoryCacheModule(FactCache, AnsiblePlugin):
+    def __init__(self, *args, **kwargs):
         self._cache = {}
-        self._cache_dir = self._get_cache_connection(cache_dir)
-        self.validate_cache_connection()
-        self._plugin = self.get_plugin(plugin_name)
+        self._display = display
+
+        plugin_name = kwargs.pop('cache_plugin')
+        plugin_path = cache_loader.find_plugin(plugin_name)
+        # FIXME Right now individual cache plugins are setting self._load_name because the line below only sets _load_name after set_options is called, which blows up
+        self._plugin = cache_loader.get(plugin_name, *args, **kwargs)
+
```

Comment:
to avoid this we shoudl be able to call set_options before running the plugin ... or we should just trigger it here as config should already be available